languages/unicode.lua

```diff
@@ -75,7 +75,7 @@ SILE.nodeMakers.unicode = pl.class({
         self:addToken(char, item)
         self:makeToken()
         self:makePenalty(0)
-      elseif self.lasttype and (self.thistype and thistype ~= lasttype and not self.isWordType[thistype]) then
+      elseif self.lasttype and (thistype and thistype ~= self.lasttype and not self.isWordType[thistype]) then
```

Comment:
But line 85 below seems wrong too, assigning a lasttype...

Reply:
And then I notice that comment above
```
-- luacheck: globals lasttype
-- XXX - this is wrong and broken, but is also confusing. See bug #687
```
So the `self.thistype` was likely wrong, but the rest is too weird to understand ^^